src/Microsoft.TemplateEngine.Cli/CommandParsing/CommandLineHelpViewerExtensions.cs

```diff
@@ -0,0 +1,111 @@
+using Microsoft.DotNet.Cli.CommandLine;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.TemplateEngine.Cli.CommandParsing
+{
+    internal static class CommandLineHelpViewerExtensions
+    {
+        private const int ColumnGutterWidth = 3;
+        private const string TemplateArgument = "TEMPLATE";
+
+        internal static string HelpView(this Command newCommand)
+        {
+            StringBuilder helpView = new StringBuilder();
+            WriteSynopsys(helpView, newCommand);
+            WriteArgumentsSection(helpView);
+            WriteOptionsSection(helpView, newCommand);
+            WriteExamples(helpView, newCommand);
+            return helpView.ToString();
+        }
+
+        private static void WriteSynopsys(StringBuilder helpView, Command command)
+        {
+            helpView.AppendLine(string.Format(LocalizableStrings.NewCommandUsage, command.Name, TemplateArgument));
+            helpView.AppendLine();
+        }
+
+        private static void WriteArgumentsSection(StringBuilder helpView)
+        {
+            helpView.AppendLine(LocalizableStrings.ArgumentsSectionTitle);
+            WriteColumnizedSummary($"  <{TemplateArgument}>", string.Format(LocalizableStrings.ArgumentDescriptionTemplate, TemplateArgument), 20, helpView);
+            helpView.AppendLine();
+        }
+
+        private static void WriteOptionsSection(StringBuilder helpView, Command command)
+        {
+            Option[] options = command
+                .DefinedOptions
+                .Where(o => !o.IsHidden())
+                .ToArray();
```

Comment:
If an array is not a must, `ToList` maybe a bit more efficient here, because the allocated memory doesn't need to exactly match the size of the data.

Reply:
Most of the members of this class are copied from [Microsoft.DotNet.Cli.CommandLine](https://github.com/dotnet/CliCommandLineParser/blob/master/src/source/CommandLine/HelpViewExtensions.cs), including the one in this and other comments. Including this code was the cost of getting proper output for dotnet new --help. I had to change implementation as Microsoft.DotNet.Cli.CommandLine do not support arguments before options and there is no way we can get proper usage with it.  However, options output is correct and should stay as is. Due to the fact that these methods (WriteOptionsSection) are private it was impossible to reuse them when changing the HelpView() to add missing Arguments description, so I had to add them to Template Engine. 
Since we are considering to move to System.CommandLine ASAP when we are back to work on Template Engine, probably this extension will go away very soon. Therefore, I'm not sure if we should invest in improving this code. 
I would like to reconsider the decision whether to refactor this class, once we get approval to include the change to 5.0.200. If won't have the approval, high chance that move to System.CommandLine will happen sooner and this code will be dropped.